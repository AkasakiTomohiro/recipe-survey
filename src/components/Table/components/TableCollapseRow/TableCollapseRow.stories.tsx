/**
 * @file TableCollapseRow.stories.tsx
 * @author Artan's Projects
 * @copyright Â© 2024 Artan's Projects. All rights reserved.
 */

import type { Meta, StoryObj } from '@storybook/react';
import { TableBody } from '@mui/material';
import Paper from '@mui/material/Paper';
import Table from '@mui/material/Table';
import TableContainer from '@mui/material/TableContainer';
import { userEvent, within } from '@storybook/testing-library';

import { TableCollapseRow } from './TableCollapseRow';

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta = {
  title     : 'components/Table/TableCollapseRow',
  component : TableCollapseRow,
  parameters: {

    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
    layout: 'centered'
  },

  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs']
} satisfies Meta<typeof TableCollapseRow>;

export default meta;
type Story = StoryObj<typeof meta>;

const decorators = [(Story: any) => (
  <TableContainer component={Paper}>
    <Table>
      <TableBody>
        <Story />
      </TableBody>
    </Table>
  </TableContainer>
)];

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args
export const Primary: Story = {
  args: {
    index: 0,
    row: ['Value1', 'Value2', 'Value3'],
    Collapse: (props) => (<>{props.index}</>)
  },
  decorators: decorators
};

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args
export const CollapseOpen: Story = {
  args: {
    index: 0,
    row: ['Value1', 'Value2', 'Value3'],
    Collapse: (props) => (<>{props.index}</>)
  },
  decorators: decorators,
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const collapseButton = canvas.getByRole('button', {});
    console.log(collapseButton)
    await userEvent.click(collapseButton);
  }
};
