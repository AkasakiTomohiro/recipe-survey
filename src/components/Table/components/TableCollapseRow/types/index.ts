/**
 * @file index.ts
 * @author Artan's Projects
 * @copyright © 2024 Artan's Projects. All rights reserved.
 */

export interface ITableCollapseRowProps {

  /**
   * 行番号
   */
  index: number;

  /**
   * 1行のデータ
   */
  row: (number | string)[];

  /**
   * Collapseで表示するコンポーネント
   */
  Collapse: (props: ICollapseProps) => JSX.Element;
}

export interface ICollapseProps {

  /**
   * 行番号
   */
  index: number
}