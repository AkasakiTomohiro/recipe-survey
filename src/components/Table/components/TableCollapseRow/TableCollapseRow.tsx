/**
 * @file TableCollapseRow.tsx
 * @author Artan's Projects
 * @copyright Â© 2024 Artan's Projects. All rights reserved.
 */

import { useState } from 'react';
import { KeyboardArrowDownIcon, KeyboardArrowUpIcon } from '~/components/Icon';

import MuiCollapse from '@mui/material/Collapse';
import IconButton from '@mui/material/IconButton';
import TableCell from '@mui/material/TableCell';
import MuiTableRow from '@mui/material/TableRow';

import { ITableCollapseRowProps } from './types';

export const TableCollapseRow = ({ row, index, Collapse }: ITableCollapseRowProps): JSX.Element => {
  const [open, setOpen] = useState(false);

  return (
    <>
      <MuiTableRow hover>
      <TableCell>
          <IconButton
            aria-label="expand row"
            size="small"
            onClick={() => setOpen(!open)}
          >
            {open ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
          </IconButton>
        </TableCell>
        {row.map((m, i) => (<TableCell key={`table-row-${index}-cell-${i}`}>{m}</TableCell>))}
      </MuiTableRow>
      <MuiTableRow>
        <TableCell style={{ paddingBottom: 0, paddingTop: 0, borderBottom: 'unset' }} colSpan={row.length + 1}>
          <MuiCollapse in={open} timeout="auto" unmountOnExit>
            <Collapse index={index} />
          </MuiCollapse>
        </TableCell>
      </MuiTableRow>
    </>
  );
};
