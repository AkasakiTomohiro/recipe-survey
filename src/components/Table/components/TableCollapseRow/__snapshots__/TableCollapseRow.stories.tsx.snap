// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Table/TableCollapseRow CollapseOpen play-test 1`] = `
<div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-xpjyhy-MuiPaper-root-MuiTableContainer-root">
  <table class="MuiTable-root css-qzpwgq-MuiTable-root">
    <tbody class="MuiTableBody-root css-apqrd9-MuiTableBody-root">
      <tr class="MuiTableRow-root MuiTableRow-hover css-dvunen-MuiTableRow-root">
        <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
          <button class="MuiButtonBase-root Mui-focusVisible MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                  aria-label="expand row"
          >
            <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                 focusable="false"
                 aria-hidden="true"
                 viewbox="0 0 24 24"
                 data-testid="KeyboardArrowUpIcon"
            >
              <path d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6z">
              </path>
            </svg>
            <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
              <span class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible MuiTouchRipple-ripplePulsate"
                    style="width: 35px; height: 35px; top: -0.5px; left: -0.5px;"
              >
                <span class="MuiTouchRipple-child MuiTouchRipple-childPulsate">
                </span>
              </span>
              <span class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
                    style="width: 35px; height: 35px; top: -0.5px; left: -0.5px;"
              >
                <span class="MuiTouchRipple-child MuiTouchRipple-childLeaving">
                </span>
              </span>
            </span>
          </button>
        </td>
        <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
          Value1
        </td>
        <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
          Value2
        </td>
        <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
          Value3
        </td>
      </tr>
      <tr class="MuiTableRow-root css-dvunen-MuiTableRow-root">
        <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root"
            colspan="4"
            style="padding-bottom: 0px; padding-top: 0px; border-bottom: unset;"
        >
          <div class="MuiCollapse-root MuiCollapse-vertical css-1fkrdiv-MuiCollapse-root"
               style="min-height: 0px; height: 20px; transition-duration: 171ms;"
          >
            <div class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper">
              <div class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner">
                0
              </div>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`components/Table/TableCollapseRow Primary smoke-test 1`] = `
<div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-xpjyhy-MuiPaper-root-MuiTableContainer-root">
  <table class="MuiTable-root css-qzpwgq-MuiTable-root">
    <tbody class="MuiTableBody-root css-apqrd9-MuiTableBody-root">
      <tr class="MuiTableRow-root MuiTableRow-hover css-dvunen-MuiTableRow-root">
        <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
          <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                  aria-label="expand row"
          >
            <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                 focusable="false"
                 aria-hidden="true"
                 viewbox="0 0 24 24"
                 data-testid="KeyboardArrowDownIcon"
            >
              <path d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z">
              </path>
            </svg>
            <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
            </span>
          </button>
        </td>
        <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
          Value1
        </td>
        <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
          Value2
        </td>
        <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
          Value3
        </td>
      </tr>
      <tr class="MuiTableRow-root css-dvunen-MuiTableRow-root">
        <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root"
            colspan="4"
            style="padding-bottom: 0px; padding-top: 0px; border-bottom: unset;"
        >
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
